### PR TITLE
Add Placement API topic

### PR DESCRIPTION
**Pull request by hhunter-ms**
## Description

Add table by @Taction to Reference/API docs as new topic

## Issue reference
#3364 

## Related PR
#3631

### CODE REVIEW

File: daprdocs/content/en/reference/api/placement_api.md [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+type: docs
+title: "Placement API reference"
+linkTitle: "Placement API"
+description: "Detailed documentation on the Placement API"
+weight: 1200
+---
+
+Dapr has an HTTP API `/placement/state` for placement service that exposes placement table information. The API is exposed on the sidecar on the same port as the healthz. This is an unauthenticated endpoint, and is disabled by default. 
+
+You need to set `DAPR_PLACEMENT_METADATA_ENABLED` environment or `metadata-enabled` command line args to true to enable it. If you are using helm, set `dapr_placement.metadataEnabled` to true.
```
Comment (msfussell) [inline daprdocs/content/en/reference/api/placement_api.md]: This need an update into the https://docs.dapr.io/reference/environment/ topic, with the Dapr sidecar as the target.

Comment (msfussell) [inline daprdocs/content/en/reference/api/placement_api.md:11]: This needs an update into the Env Variables doc topic

Comment (Taction) [reply daprdocs/content/en/reference/api/placement_api.md]: One more thing, the environment variable is for Dapr placement, so the target would be Dapr placement.

Comment (hhunter-ms) [reply daprdocs/content/en/reference/api/placement_api.md]: Pushed this update - thanks!

File: daprdocs/content/en/reference/api/placement_api.md [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+
+## Examples
+
+```shell
+ curl localhost:8080/placement/state
+```
+
+```json
+{
+	"hostList": [{
```
Comment (msfussell) [inline daprdocs/content/en/reference/api/placement_api.md:58]: Just replace this with the same changes from Taction, showing multiple actors

File: daprdocs/content/en/reference/api/placement_api.md (hunk 1 above)
Comment (msfussell) [inline daprdocs/content/en/reference/api/placement_api.md]: Actually this needs more detail. 
@Taction
1) Is this really just called "metadata-enabled" and not ""placement-metadata-enabled""
2) Also `metadata-enabled` needs to be added to the https://docs.dapr.io/reference/arguments-annotations-overview/ . When you say command line is this on daprd or the CLI

File: daprdocs/content/en/reference/environment/_index.md [hunk 3]
```diff
@@ -26,4 +26,5 @@ The following table lists the environment variables used by the Dapr runtime, CL
...
 | OTEL_EXPORTER_OTLP_PROTOCOL | OpenTelemetry Tracing | The OTLP protocol to use Transport protocol. (`grpc`, `http/protobuf`, `http/json`) |
 | DAPR_COMPONENTS_SOCKETS_FOLDER | Dapr runtime and the .NET, Go, and Java pluggable component SDKs | The location or path where Dapr looks for Pluggable Components Unix Domain Socket files. If unset this location defaults to `/tmp/dapr-components-sockets` |
-| DAPR_COMPONENTS_SOCKETS_EXTENSION | .NET and Java pluggable component SDKs | A per-SDK configuration that indicates the default file extension applied to socket files created by the SDKs. Not a Dapr-enforced behavior. |
\ No newline at end of file
+| DAPR_COMPONENTS_SOCKETS_EXTENSION | .NET and Java pluggable component SDKs | A per-SDK configuration that indicates the default file extension applied to socket files created by the SDKs. Not a Dapr-enforced behavior. |
+| DAPR_PLACEMENT_METADATA_ENABLED | Dapr placement | A placement service that exposes placement table information. Set to `true` to enable. [Learn more about the Placement API]({{< ref placement_api.md >}}) |
```
Comment (msfussell) [inline daprdocs/content/en/reference/environment/_index.md]:
```suggestion
| DAPR_PLACEMENT_METADATA_ENABLED | Dapr placement | Enable an endpoint for the Placement service that exposes placement table information on actor usage. Set to `true` to enable. [Learn more about the Placement API]({{< ref placement_api.md >}}) |
```

File: daprdocs/content/en/reference/api/placement_api.md (hunk 1 above)
Comment (Taction) [reply daprdocs/content/en/reference/api/placement_api.md]: This is the environment or command line args for the `placement` binary not for daprd or cli, and this is supposed to only be used in self-host mode(In k8s mode, the helm value should be used). I've found this doc https://docs.dapr.io/operations/hosting/self-hosted/self-hosted-no-docker/#enable-actors  introduces how to run placement manually, maybe add a sub-section for supported  `arguments and environments` here.

File: daprdocs/content/en/reference/api/placement_api.md [hunk 4]
```diff
@@ -0,0 +1,77 @@
...
+type: docs
+title: "Placement API reference"
+linkTitle: "Placement API"
+description: "Detailed documentation on the Placement API"
+weight: 1200
+---
+
+Dapr has an HTTP API `/placement/state` for placement service that exposes placement table information. The API is exposed on the sidecar on the same port as the healthz. This is an unauthenticated endpoint, and is disabled by default. 
+
+You need to set `DAPR_PLACEMENT_METADATA_ENABLED` environment or `metadata-enabled` command line args to true to enable it. If you are using helm, set `dapr_placement.metadataEnabled` to true.
```
Comment (msfussell) [inline daprdocs/content/en/reference/api/placement_api.md]:
```suggestion
To enable the placement metadata in self-hosted mode you can either set`DAPR_PLACEMENT_METADATA_ENABLED` environment variable or `metadata-enabled` command line args on the Placement service to `true` to. See [how to run the Placement service in self-hosted mode]({{< ref "self-hosted-no-docker/#enable-actors >}})
If you are using Helm for deployment of the Placement service on Kubernetes then to enable the placement metadata, set `dapr_placement.metadataEnabled` to `true`.
```

File: daprdocs/content/en/reference/api/placement_api.md (hunk 1 above)
Comment (msfussell) [reply daprdocs/content/en/reference/api/placement_api.md]: I see this was not clear. I have made a suggestion below for an update. Please review

File: daprdocs/content/en/reference/environment/_index.md [hunk 5]
```diff
@@ -26,4 +26,5 @@ The following table lists the environment variables used by the Dapr runtime, CL
...
 | OTEL_EXPORTER_OTLP_PROTOCOL | OpenTelemetry Tracing | The OTLP protocol to use Transport protocol. (`grpc`, `http/protobuf`, `http/json`) |
 | DAPR_COMPONENTS_SOCKETS_FOLDER | Dapr runtime and the .NET, Go, and Java pluggable component SDKs | The location or path where Dapr looks for Pluggable Components Unix Domain Socket files. If unset this location defaults to `/tmp/dapr-components-sockets` |
-| DAPR_COMPONENTS_SOCKETS_EXTENSION | .NET and Java pluggable component SDKs | A per-SDK configuration that indicates the default file extension applied to socket files created by the SDKs. Not a Dapr-enforced behavior. |
\ No newline at end of file
+| DAPR_COMPONENTS_SOCKETS_EXTENSION | .NET and Java pluggable component SDKs | A per-SDK configuration that indicates the default file extension applied to socket files created by the SDKs. Not a Dapr-enforced behavior. |
+| DAPR_PLACEMENT_METADATA_ENABLED | Dapr placement | Enable an endpoint for the Placement service that exposes placement table information on actor usage. Set to `true` to enable. [Learn more about the Placement API]({{< ref placement_api.md >}}) |
```
Comment (msfussell) [inline daprdocs/content/en/reference/environment/_index.md]:
```suggestion
| DAPR_PLACEMENT_METADATA_ENABLED | Dapr placement | Enable an endpoint for the Placement service that exposes placement table information on actor usage. Set to `true` to enable in self-hosted mode. [Learn more about the Placement API]({{< ref placement_api.md >}}) |
```

**Comment by msfussell** [comment]
@hhunter-ms - One small issue. Can we link from this article https://v1-12.docs.dapr.io/concepts/dapr-services/placement/#usecase to this API doc?

**Review by msfussell — APPROVED** [review]
LGTM

**Pull request merged**